Large Broadcom Ports Folder

### CircuitPython version
7.0.0

### Code/REPL

```python
git submodule update --init
```


### Behavior

When running `git submodule update --init`, the size is >8gb just for `circuitpython/ports/broadcom/firmware`

Is there a way to target a different repo or branch when building?

Long module sync time

```


### Description

_No response_

### Additional information

_No response_

Thanks Scott! It wasn't only the speed, I was pulling my hair out trying to free up space on my little build PIs, got up to 15G but still wasn't enough.

Anyway, I'm really clueless when it comes to make so I"m sure I'm doing something stupid. This is what I get when I try the make fetch-submodules:

>  make fetch-submodules BOARD=adafruit_feather_esp32s2
Use make V=1, make V=2 or set BUILD_VERBOSE similarly in your environment to increase build verbosity.
find: ‘../../extmod/ulab/code’: No such file or directory
make: *** No rule to make target 'fetch-submodules'.  Stop.

--- a/.gitmodules
+++ b/.gitmodules
@@ -198,3 +198,4 @@
 	path = ports/broadcom/firmware
 	url = https://github.com/raspberrypi/rpi-firmware.git
 	branch = master
+	shallow = true

--- a/.gitmodules
+++ b/.gitmodules
@@ -109,7 +109,7 @@
 	url = https://github.com/adafruit/Adafruit_CircuitPython_Register.git
 [submodule "extmod/ulab"]
 	path = extmod/ulab
-	url = https://github.com/adafruit/circuitpython-ulab
+	url = https://github.com/v923z/micropython-ulab
 [submodule "frozen/Adafruit_CircuitPython_ESP32SPI"]
 	path = frozen/Adafruit_CircuitPython_ESP32SPI
 	url = https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI

--- a/Makefile
+++ b/Makefile
@@ -321,3 +321,8 @@ clean-nrf:
 
 clean-stm:
 	$(MAKE) -C ports/stm BOARD=feather_stm32f405_express clean
+
+.PHONY: fetch-submodules
+fetch-submodules:
+	git submodule update --init -N --depth 1
+	git submodule foreach 'git fetch --tags --depth 1 origin $$sha1 && git checkout -q $$sha1'

--- a/Makefile
+++ b/Makefile
@@ -324,5 +324,8 @@ clean-stm:
 
 .PHONY: fetch-submodules
 fetch-submodules:
-	git submodule update --init -N --depth 1
+	# This update will fail because the commits we need aren't the latest on the
+	# branch. We can ignore that though because we fix it with the second command.
+	# (Only works for git servers that allow sha fetches.)
+	git submodule update --init -N --depth 1 || true
 	git submodule foreach 'git fetch --tags --depth 1 origin $$sha1 && git checkout -q $$sha1'

